src/grid.js

```diff
@@ -1,21 +1,36 @@
 // @flow
 import React from 'react'
 import PropTypes from 'prop-types'
-import { compact } from 'lodash'
+import { compact, omit } from 'lodash'
 import { getSpacingClasses } from './utils'
-import type { Dev, AlignGrid, Justify, Size, Spacing } from './types'
+import type {
+  Dev,
+  AlignGrid,
+  Justify,
+  Size,
+  Spacing,
+  SpacingValues,
+} from './types'
 import styles from './grid.css'
 import devStyles from './dev.css'
 
 export type Props = {
+  margin: Spacing,
   dev?: Dev,
   align?: AlignGrid,
   children?: React$Element<any> | Array<React$Element<any>>,
   className?: string,
   justify?: Justify,
-  margin?: Spacing,
   padding?: Spacing,
   size?: Size,
+  marginTop?: SpacingValues,
+  marginRight?: SpacingValues,
+  marginBottom?: SpacingValues,
+  marginLeft?: SpacingValues,
+  paddingTop?: SpacingValues,
+  paddingRight?: SpacingValues,
+  paddingBottom?: SpacingValues,
+  paddingLeft?: SpacingValues,
```

Comment:
this is reeeally nit picky but you could sort them alphabetically with something like https://github.com/Tyriar/vscode-sort-lines